### PR TITLE
Add Role of OGC API Records in Overview chapter (#171)

### LINKED ISSUE #171: Finish chapter 6 Overview
Chapter 6 of OGC API Records Core seems to still need some work: 
- some headings start with "tl;dr", this needs to be addressed
- there is a Note at the start about the need to add a discussion about migration from big repositories of metadata to distributed crawlable catalogues
- I suggest reordering the chapter. Start with what is now 6.4 General and 6.5 Building Blocks, integrate 6.1-6.3 into what is now 6.5, then the rest.
- the second and third paragraph of 6.8.2. API Behaviour Model could perhaps be deleted. As a reader I don't find they are necessary / don't contain need to know information.
- Tables 5 and 6 in chapter 6 are repeated in chapter 7 and further, in the conformance classes. In chapter 6 I think just a summary and a reference to the full table in the conformance class would be enough, e.g. "he core set of properties that may be used to describe a record collection or catalogue include id, title and description. See table xx for the full set."

### PR DESCRIPTION
**Pull request by kalxas**
Closes #171

### CODE REVIEW

**Review by pvretano — APPROVED** [review]

File: core/standard/clause_6_overview.adoc [hunk 1]
```diff
@@ -1,7 +1,14 @@
 [[overview]]
 == Overview
 
-NOTE: Need to have a disucssion in here about the migration of catalogues from big repositories of metadata documents (e.g. ISO19115) to distributed, crawlable catalogies (e.g. static STAC) and harvesting of resource into the catalogue to make them searchable (e.g. harvesting records close to resources (i.e. STAC item) or even harvesting the resource itself (e.g. the collection object)).
+[[records-role]]
+=== Role of OGC API Records
```
Comment (tomkralidis) [inline core/standard/clause_6_overview.adoc:5]: `s/OGC API Records/OGC API - Records/g`

File: core/standard/clause_6_overview.adoc [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+
+Historically, catalogues have been used as big repositories of formal metadata like ISO-19119, ISO-19115, FGDC. etc. However, more recently catalogues are migrating to a more distributed model where the metadata is intimately associated with the resource that it describes. A modern example is STAC which relies on well defined building blocks that can be linked in different ways.
+
+Althought OGC API - Records is built upon the legacy of CSW, its role in a modern SDI can be both: a big repository of metadata documents or a distributed (even crawlable) catalogue.
```
Comment (tomkralidis) [inline core/standard/clause_6_overview.adoc:9]: Although

File: core/standard/clause_6_overview.adoc [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
+
+In the modern scenario, the catalogue records can be distributed and linked together so that they can be crawled OR they can be harvested into a service endpoint to make them searchable. Even more than that, a searchable catalogue can harvest not only a metadata document describing the resource but in some cases the resource itself (e.g. harvesting the collections object from OGC API). The implications of this are subtle but important. In the traditional cataloging approach (e.g. based on CSW) a client could find the record and the record might include binding information to access the resource but that is pretty much it. Using the new approach, a client interacts more intimately with the catalogue and the catalogue becomes integral to the access to the resource. Think of the entire cloud native ecosystem: range requests to COGs don't really work unless there is a STAC item telling the client in what ranges the bands exists, thus the catalogue becomes an integral component in the process of accessing a resource.
```
Comment (tomkralidis) [inline core/standard/clause_6_overview.adoc:11]: `s/Even more than that/In addition/`

Comment (tomkralidis) [reply core/standard/clause_6_overview.adoc:11]: As well:
Replace:
Think of the entire cloud native ecosystem: range requests to COGs don't really work unless there is a STAC item telling the client in what ranges the bands exists, thus the catalogue becomes an integral component in the process of accessing a resource.

with

An example is the cloud native ecosystem: HTTP range requests to Cloud Optimized GeoTIFF (COG) data cannot be used propertly unless there is a STAC item telling the client in what ranges the bands exists, thus the catalogue becomes an integral component in the process of accessing a resource.

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: core/standard/clause_6_overview.adoc (hunk 2 above)
Comment (tomkralidis) [reply core/standard/clause_6_overview.adoc:9]: Update to:

Historically, catalogues have been used as large repositories of formal metadata like ISO 19119, ISO 19115, FGDC, etc. However, more recently catalogues are migrating to a more distributed model where the metadata is intimately associated with the resource that it describes. A modern example is STAC which relies on well defined building blocks that can be linked in different ways.

File: core/standard/clause_6_overview.adoc (hunk 1 above)
Comment (kalxas) [reply core/standard/clause_6_overview.adoc]: Fixed, thanks

File: core/standard/clause_6_overview.adoc (hunk 2 above)
Comment (kalxas) [reply core/standard/clause_6_overview.adoc]: Fixed, thanks

File: core/standard/clause_6_overview.adoc (hunk 3 above)
Comment (kalxas) [reply core/standard/clause_6_overview.adoc]: Fixed, thanks

**Pull request merged**